core/src/main/scala/org/apache/spark/executor/CoarseGrainedExecutorBackend.scala

```diff
@@ -479,6 +479,32 @@ private[spark] object CoarseGrainedExecutorBackend extends Logging {
       driverConf.set(EXECUTOR_ID, arguments.executorId)
       cfg.logLevel.foreach(logLevel => Utils.setLogLevelIfNeeded(logLevel))
 
+      // Set executor memory related config here according to resource profile
+      if (cfg.resourceProfile.id != ResourceProfile.DEFAULT_RESOURCE_PROFILE_ID) {
+        cfg.resourceProfile
+          .executorResources
+          .foreach {
+            case (ResourceProfile.OFFHEAP_MEM, request) =>
+              driverConf.set(MEMORY_OFFHEAP_SIZE.key, request.amount.toString + "m")
+              if (request.amount > 0) {
+                driverConf.set(MEMORY_OFFHEAP_ENABLED.key, "true")
```

Comment:
this is questionable to me, we haven't supported setting confs yet so I would expect it to pick up the default config for this.  I'd rather see a separate issue if we want to revisit this behavior.

Reply:
> this is questionable to me, we haven't supported setting confs yet so I would expect it to pick up the default config for this. I'd rather see a separate issue if we want to revisit this behavior.

Make sense to me. I'll address it later, how about introducing a config to control whether to enable offheap enabled or not?

Reply:
> Thank you very much for this PR, we just encountered this problem. Is the logic of resetting the offheap configuration better this way?
> 
> ```
> case (ResourceProfile.OFFHEAP_MEM, request)  if  request.amount > 0 =>
>        driverConf.set(MEMORY_OFFHEAP_SIZE.key, request.amount.toString + "m")
>        driverConf.set(MEMORY_OFFHEAP_ENABLED.key, "true")
> ```

I think this is incorrect. Imagine such a scenario where default resource profile's offheap is 512M, and resource profile with id 2 's offheap is 0, in such case the offheap config will be incorrect!